.changeset/nice-impalas-whisper.md

```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/cloudflare': patch
+---
+
+Fixes regression in Astro asset's image service
```

Comment:
```suggestion
Fixes a regression in `@astrojs/cloudflare` adapter where it would overwrite the image configuration even when compatible with Cloudflare Images. The adapter now only overwrites the configuration when known to be incompatible, and has moved the compatibility logic into the adapter itself.
```